modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/alert/AlertDefinitionManagerBean.java

```diff
@@ -343,6 +348,10 @@ public int removeAlertDefinitions(Subject subject, int[] alertDefinitionIds) {
         criteria.addFilterIds(ArrayUtils.wrapInArray(alertDefinitionIds));
         criteria.addFilterDeleted(false);
         criteria.clearPaging();
+
+        if ( groupId != null ){
+            criteria.addFilterResourceGroupIds(groupId);
```

Comment:
Does this actually work? From what I gather, it does not change anything. When I select an alert definition from a group's alert definition page the alert id is the group alert definition. The `groupId` is not relevant.

Reply:
It supposes to not be relevant, but the thing is how the AlertDefinitionCriteria works, according to this: 

https://github.com/rubenvp8510/rhq/blob/8d1ede47c80394e9bc7a587d21c1ed6b3baf9417/modules/core/domain/src/main/java/org/rhq/core/domain/criteria/AlertDefinitionCriteria.java#L37-L37

I need to set the ```groupId``` in order to get the AlertDefinitions,  if I don't do it the query doesn't return anything.  This is why for the cases on the users that don't have Inventory Manager permissions but have Alert Definition permissions, the alert is not deleted, because of this:

https://github.com/rubenvp8510/rhq/blob/03438c5b843693a26f29d0c853d47eea4b8b7635/modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/alert/AlertDefinitionManagerBean.java#L355-L355

Returns 0.

Reply:
I could modify this, but I would prefer to not do it.

Reply:
So will the delete cascade then? I think that was what my concern was. Thanks for the clarification.

Reply:
The groupId is just for permission validations, as the comment on AlertDefinitionCriteria said.